### PR TITLE
Feature/dst notification integration

### PR DESCRIPTION
**Pull request by akhil-ta**

### CODE REVIEW

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+                scheduleDSTBroadCast(context, dst_end_request_code, endCalendar);
+                dst_end_request_code++;
+            } else {
+                scheduleDSTBroadcastForNextYear(context);
+            }
+            resetNotificationDateInLibrary(context);
+        }
+    }
+
+    private void resetNotificationDateInLibrary(Context context) {
```
Comment (kamalkkishoreta) [inline android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java]: Can we change the function name? This is already a library.

Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java]: Sorry, I didn't understand that. did you mean the function is in the library so we need to change that right?  if yes what should be the name? can we change it to resetNotificationDate?

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/TimezoneChangeReceiver.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+
+public class TimezoneChangeReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        final String action = intent.getAction();
+        if (action.equals(Intent.ACTION_TIMEZONE_CHANGED) || action.equals(Intent.ACTION_TIME_CHANGED)) {
```
Comment (kamalkkishoreta) [inline android/src/main/java/com/dieam/reactnativepushnotification/dst/TimezoneChangeReceiver.java:12]: Are we not considering DATE_CHANGED?

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java (hunk 1 above)
Comment (kamalkkishoreta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/DSTManager.java]: resetNotificationDate should be fine!

**Review by kamalkkishoreta — APPROVED** [review]

**Review by kamalkkishoreta — APPROVED** [review]

File: android/src/main/java/com/dieam/reactnativepushnotification/dst/TimezoneChangeReceiver.java (hunk 2 above)
Comment (akhil-ta) [reply android/src/main/java/com/dieam/reactnativepushnotification/dst/TimezoneChangeReceiver.java:12]: DATE_CHANGED call back is not firing, only ACTION_TIME_CHANGED is firing. we only need a callback when a user updates the date. ACTION_TIME_CHANGED will fire if a user changes the system time/date.

**Pull request merged**